server/datastore/mysql/apple_mdm.go

```diff
@@ -2238,6 +2241,9 @@ func (ds *Datastore) RecordHostBootstrapPackage(ctx context.Context, commandUUID
 }
 
 func (ds *Datastore) GetHostMDMMacOSSetup(ctx context.Context, hostID uint) (*fleet.HostMDMMacOSSetup, error) {
+	// NOTE: consider joining on host_dep_assignments instead of host_mdm so DEP hosts that
+	// manually enroll or re-enroll are included in the results so long as they are not unassigned
+	// in Apple Business Manager
```

Comment:
here and above: we might have already discussed this somewhere but the problem with using `host_dep_assignments` is that a host can be assigned to Fleet in ABM but still manually enroll.

we should probably keep using `host_mdm`, but be better at updating the table with the right values when a host enrolls (we spitballed adding a query param)